sled-agent/src/rack_setup/service.rs

```diff
@@ -515,13 +570,18 @@ impl ServiceInner {
         futures::future::join_all(plan.iter().map(
             |(_, allocation)| async move {
                 let sled_address = SocketAddr::V6(get_sled_address(
-                    *allocation.initialization_request.subnet.as_ref(),
+                    allocation.initialization_request.subnet.as_ref().0,
                 ));
-                self.initialize_services(
-                    sled_address,
-                    &allocation.services_request.services,
-                )
-                .await?;
+
+                let all_services = allocation
+                    .services_request
+                    .services
+                    .iter()
+                    .chain(allocation.services_request.dns_services.iter())
```

Comment:
I think part of my confusion in the above comment about the number of sleds asked to run DNS comes from this line. This seems to send the requests to setup DNS again, and also to all sleds. I think that request is idempotent (though I could be wrong), but I'm not sure what to make of it.

Reply:
The API to `services_request` is basically "please give me all the services you'd like me to be running".

So if we ask for: "A", it'll run "A". If we ask for "A, B, C", it'll run "A, B, C"

However, if we ask for "A", and then later ask for "B, C", it'll notice that "A" was no longer requested, and will throw an error (although eventually, I'd like it to actually stop running "A"!). This API should let Nexus "ensure" the set of services that should be running on a sled at basically anytime - and if it's the same set, it should be idempotent.